### PR TITLE
[Enhancement] Support manual add backend to blacklist

### PR DESCRIPTION
**Pull request by Seaven**
Why I'm doing:

What I'm doing:

* update auto blacklist strategies
  * record all disconnected event, If there are too many disconnections in the term, don't removed from the blacklist
* add manual add backend to blacklist stynax
* blacklist only in memory, and each FE manages independently

```
MySQL td> add backend blacklist 3609624;
Query OK, 0 rows affected
Time: 0.002s
MySQL td> show backend blacklist
+-----------+------------------+-------------------------------+----------------------------+---------------+
| BackendId | AddBlackListType | LostConnectionTime            | LostConnectionNumberInTerm | CheckTimeTerm |
+-----------+------------------+-------------------------------+----------------------------+---------------+
| 3609624   | MANUAL           | 2024-01-15T19:14:57.962644665 | 0                          | 5             |
+-----------+------------------+-------------------------------+----------------------------+---------------+
1 row in set
Time: 0.005s
MySQL td> delete backend blacklist 3609624;
Query OK, 0 rows affected
Time: 0.001s
MySQL td>
MySQL td> show backend blacklist
+-----------+------------------+--------------------+----------------------------+---------------+
| BackendId | AddBlackListType | LostConnectionTime | LostConnectionNumberInTerm | CheckTimeTerm |
+-----------+------------------+--------------------+----------------------------+---------------+
+-----------+------------------+--------------------+----------------------------+---------------+
0 rows in set
Time: 0.007s
MySQL td>
```

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by packy92** [comment]
please add a privilege check to this add/delete clause.

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -1815,6 +1820,20 @@ showWhiteListStatement
     : SHOW WHITELIST
     ;
 
+// ------------------------------------ backend BlackList Statement ---------------------------------------------------
+
+addBackendBlackListStatement
+    : ADD BACKEND BLACKLIST INTEGER_VALUE
```
[the remark this answers is not in the document]
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1826]: OK，let met modify

**Comment by liuyehcf** [comment]
Good feature for testing!

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 2]
```diff
@@ -2781,4 +2781,10 @@ public class Config extends ConfigBase {
 
     @ConfField(mutable = true)
     public static long jdbc_meta_default_cache_expire_sec = 600L;
+
+    @ConfField(mutable = true)
+    public static long black_host_history_sec = 2 * 60; // 2min
+
+    @ConfField(mutable = true)
+    public static long black_host_connect_failures_within_time = 5;
```
Comment (liuyehcf) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:2789]: Better adding some comments to describe

**Review by packy92 — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 3]
```diff
@@ -419,6 +420,8 @@ public ShowResultSet execute() throws AnalysisException, DdlException {
             handleShowFailPoint();
         } else if (stmt instanceof ShowDictionaryStmt) {
             handleShowDictionary();
+        } else if (stmt instanceof ShowBackendBlackListStmt) {
+            handleBackendBlackList();
```
Comment (liuyehcf) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:424]: will this command pass to leader if served by follower?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:424]: no, need execute on each FE

**Review by packy92 — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Pull request merged**